### ISSUE TITLE
Internal error details are leaked

### ISSUE THREAD

**Issue by mattias-p** [zonemaster/zonemaster-backend#213]
When an unhandled error occurs in the web backend an error reponse is given with error code -32603. For example if the `backend_config.ini` file is missing I get this response:
```json
{"jsonrpc":"2.0","id":"140715758026879","error":{"code":-32603,"message":"UNABLE TO LOAD /etc/zonemaster/backend_config.ini","code":-32603},"id":"140715758026879"}
```
This leaks details about the internals of the service. Instead this error message should be logged and the message in the response should be simply `"Internal error"` (as prescribed by the JSON-RPC 2.0 specification).

**Comment by mattias-p** [comment]
This has not been fixed.

**Comment by matsduf** [comment]
Issue #214 must be fixed to fix this since we need to send error messages to a log file.

**Comment by mattias-p** [comment]
I had things mixed up when I spoke earlier. RPCAPI does have a log file. #314 seems to be about the volume and quality of the log file.

**Comment by mtoma** [comment]
Not really sure about the perimeter of what needs to be done here.
My idea was to catch the JSON::RPC messages and redirect them to the log file and replace with "Internal Error" sent to the user.
Can't fix right now because search.cpanorg returns "Internal Server Error" right now.

**Comment by mattias-p** [comment]
Maybe we could have a special wrapper for user-facing error messages and replace everything else with "Internal Server Error"? Then we can start adding that wrapper on a case-by-case basis. How does that sound?

I'm using metacpan.org where I can while search.cpan.org is down.

**Comment by mtoma** [comment]
I don't really see a way to catch the JSON::RPC errors in any way:
http://search.cpan.org/~dmaki/JSON-RPC-1.06/lib/JSON/RPC.pm#ERRORS

The only option seems to me to modify the code of the JSON::RPC module itself or to add and eval to each of the methods of Backend PM and catch the errors before JSON RPC does.

While this is feasible it seems to me we should consider if we shall use the JSON::RPC module at all as we no longer use any of its functionalities. We validate the arguments outside of the module. If we also catch all crashes what is the use of JSON::RPC then?

**Comment by sandoche2k** [comment]
@mtoma @mattias-p this issue exists from 2016. Can any of you have a look?

**Comment by mattias-p** [comment]
Well, there is #545. I reviewed that in October last year. I'm not sure what more I can do at this point.

**Comment by matsduf** [comment]
I suggest that we consider this issue to be resolved when #545 is updated and merged.
